### PR TITLE
White and dark tiles craftable 2: Electric Boogalo!

### PR DESCRIPTION
**Pull request by asperger-sind**
## About the PR 
Adds white and dark tiles to the crafting menu.
Basically a copypaste of the steel tile recipe BUT with white/dark tiles instead of steel tiles.

**Changelog**

:cl: ninruB
- add: a recipe for white and dark tiles in the crafting menu, sorta like the steel tile recipe.

### CODE REVIEW

**Comment by asperger-sind** [comment]
Weird, checking the failed build & test I see this abomination ` Error Message:
   System.Collections.Generic.KeyNotFoundException : The given key 'woodtile' was not present in the dictionary.`
   even considering that I didn't touch the woodtile part whatsoever.

File: Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml [hunk 1]
```diff
@@ -28,5 +28,37 @@
             # Needs StackType ID
             - material: WoodPlank
               amount: 1
-    - node: woodtile
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml:31]: You sure bud? Has this removed the woodtile node which is why you got the error.

File: Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml [hunk 2]
```diff
@@ -28,5 +28,37 @@
...
+    - node: start
+      edges:
+        - to: whitetile
+          completed:
+            - !type:SetStackCount
+              amount: 4
+          steps:
+            - material: Steel
+              amount: 1
+    - node: steeltile
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml:47]: This is node whitetile yes?

File: Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml [hunk 3]
```diff
@@ -28,5 +28,37 @@
             # Needs StackType ID
             - material: WoodPlank
               amount: 1
-    - node: woodtile
+    - node: whitetile
       entity: FloorTileItemWood
+
+- type: constructionGraph
+  id: tileWhite
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml:35]: All IDs should ideally be PascalCase so TileWhite.

**Review by Peptide90 — CHANGES_REQUESTED** [review]
Left some comments, reviewed on mobile but think I can see why you had issues with wood tiles.

File: Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml (hunk 1 above)
Comment (asperger-sind) [reply Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml:31]: Either I'm totally blind and can't read changelogs of stuff I made or you're right.

File: Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml [hunk 4]
```diff
@@ -30,3 +30,35 @@
...
+            - !type:SetStackCount
+              amount: 4
+          steps:
+            - material: Steel
+              amount: 1
+    - node: whitetile
+      entity: FloorTileItemWhite
+
+- type: constructionGraph
+  id: tileDark
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Crafting/Graphs/tiles.yml:51]: Nearly there :P TileDark here

File: Resources/Prototypes/Recipes/Crafting/tiles.yml [hunk 5]
```diff
@@ -22,28 +22,28 @@
...
-#   startNode: start
-#   targetNode: darktile
-#   category: Tiles
-#   description: "A dark station tile."
-#   icon: Objects/Tiles/tile.rsi/dark.png
-#   objectType: Item
-#
+- type: construction
+  name: white tile
+  id: tileWhite
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Crafting/tiles.yml:27]: ID's now need to match what's in the graphs so TileWhite

File: Resources/Prototypes/Recipes/Crafting/tiles.yml [hunk 6]
```diff
@@ -22,28 +22,28 @@
...
+  startNode: start
+  targetNode: whitetile
+  category: Tiles
+  description: "Four white station tiles."
+  icon: Objects/Tiles/tile.rsi/white.png
+  objectType: Item
+
+- type: construction
+  name: dark tile
+  id: tileDark
```
Comment (Peptide90) [inline Resources/Prototypes/Recipes/Crafting/tiles.yml:38]: TileDark

**Review by Peptide90 — CHANGES_REQUESTED** [review]
In addition to the comments made, you'll also then need to add the following to the tiles so that they actually construct too:

1. Head to entities/objects/misc/tiles.yml
2. Go down to the tile entity you want, in this case lines 57 and 73 onwards.
3. Add the following code to the bottom of each entity:
`  - type: Construction
    graph: TileWhite
    node: whitetile`
And the same thing but for TileDark. Make sure the graph matches the ID and the node matches the end point node.

If you'd like you can also update the other ID's both in the graph, in the tiles.yml crafting menu and in the tiles.yml object file, only if you want. That way they all conform to conventions.

Good work so far, you'll get there, just takes some knowing where things are. Your next PR tests will run automatically to catch all this stuff and make it easier.

**Comment by asperger-sind** [comment]
Pretty sure this PR works now and since Vera is on break can someone instead of Vera merge this kthnx

**Pull request merged**